scripts/aggregate-themes/theme-aggregator.js

```diff
@@ -156,6 +91,84 @@ class ThemeAggregator {
     return assets;
   }
 
+  /**
+   * Triggers aggregation and generation for a given theme.
+   * @param {string} themeName - The theme to aggregate.
+   * @param {Object} options - The aggregation options.
+   * @param {boolean} defaultFlag - Whether the theme to be generated is a root or scope theme. Guards against default theme and scope theme being equivalent.
+   * @returns {array} - An array of file names.
+   */
+  static triggerAggregationAndGeneration(themeName, options = {}, defaultFlag) {
+    if (!themeName) return null;
+    Logger.log(`Aggregating ${themeName} files...`, { LOG_CONTEXT });
+    const { theme, scoped = [] } = options;
+    const isRoot = themeName === theme && defaultFlag;
+    const isScoped = scoped.indexOf(themeName) > -1;
+    const themeScope = { isRoot, isScoped };
+
+    const aggregatedAssets = ThemeAggregator.aggregateTheme(themeName, themeScope, options);
+    const generatedAsset = ThemeAggregator.generateTheme(themeName, themeScope, options);
+    // Generated theme file should take precedence over aggregated files.
+    // Therefore, take advantage of css import precdence by adding the generated asset last.
```

Comment:
I'm confused, what is this the difference between the aggregated theme and the generated theme? I would have thought these were conceptually the sam thing, only placing the imports in one file?

Reply:
yeah we have an issue with semantics. We "aggregate" root and scoped theme files, "aggregate" theme-name files then "generate" root or scoped theme files. The aggregated root/scoped theme files and generated root/scoped theme files are then imported into the js file. 

Maybe I should reword "aggregated files" to "existing root and scoped theme files" here?